### PR TITLE
(GH-1999) Unexperimentalify project docs

### PR DESCRIPTION
**Pull request by hestonhoffman**
Moves Bolt project docs out of experimental state.
- Combined + edited project and project directory pages.
- Added a topichead for Configuring Bolt. This allows us to move the
  Configuring Bolt doc down in the hierarchy so that projects and
  inventory files are introduced first.
- Adds experimental 🧪 and deprecated ⛔️ symbols to
  relevant sections of the docs.

!no-release-note

### CODE REVIEW

**Comment by hestonhoffman** [comment]
This is a pretty big one 😬

**Comment by nicklewis** [comment]
"This" being the word "unexperimentalify"? 😉

File: documentation/projects.md [hunk 1]
```diff
@@ -49,29 +25,41 @@ To get started with a Bolt project:
...
 
-The `bolt-project.yaml` file holds options to configure your Bolt project, as
-well as options to control how Bolt behaves when you run a project.
+Besides the `name` key, the `bolt-project.yaml` file holds options to configure
+your Bolt project, as well as options to control how Bolt behaves when you run a
+project.
 
-For example, if you wanted Bolt to load a custom module path, you could use the
-following in your `bolt-project.yaml`:
+For example, if you want to add a little flare to your Bolt output, use the
```
Comment (beechtom) [inline documentation/projects.md:52]:
```suggestion
For example, if you want to add a little flair to your Bolt output, use the
```

I don't know if we want people to light their Bolt output on fire 😉

File: documentation/templates/bolt_project_reference.md.erb [hunk 2]
```diff
@@ -48,3 +45,7 @@ which is supported in the [project directory](configuring_bolt.md).
 # bolt-project.yaml
 <%= @yaml.to_yaml(indentation: 2).split("\n").drop(1).join("\n") %>
 ```
+
+📖 **Related information**  
+
+- [Bolt projects](projects.md)
```
Comment (beechtom) [inline documentation/templates/bolt_project_reference.md.erb:51]:
```suggestion
- [Bolt projects](projects.md)

```

File: documentation/projects.md (hunk 1 above)
Comment (hestonhoffman) [reply documentation/projects.md:52]: Rofl. Good catch

**Pull request merged**